### PR TITLE
Replaced with cap_notify_coauthors

### PR DESCRIPTION
**Pull request by rebeccahum**
As per issue #403, new function cap_notify_coauthors hooks onto comment_notification_recipients filter and replaces old custom function wp_notify_postauthor.  

Note: This would render PR #412 (issue #411) unnecessary.

### CODE REVIEW

File: co-authors-plus.php [hunk 1]
```diff
@@ -1540,118 +1543,46 @@ public function clear_cache_on_terms_set( $object_id, $terms, $tt_ids, $taxonomy
...
-				}
-			}
+ 			// The comment was left by the co-author
+ 			if ( $comment->user_id == $author->ID ) {
+ 				continue;
+ 			}
 
-			$message_headers = "$from\n"
-				. 'Content-Type: text/plain; charset="' . get_option( 'blog_charset' ) . "\"\n";
+ 			// The co-author moderated a comment on his own post
```
Comment (mjangda) [inline co-authors-plus.php]: Let's change "his" to the more gender neutral "their'

File: co-authors-plus.php [hunk 2]
```diff
@@ -1540,118 +1543,46 @@ public function clear_cache_on_terms_set( $object_id, $terms, $tt_ids, $taxonomy
...
+ 			if ( $author->ID == get_current_user_id() ) {
+ 				continue;
+ 			}
 
-			if ( isset( $reply_to ) ) {
-				$message_headers .= $reply_to . "\n";
-			}
+ 			// If there's no email to send the comment to
+ 			if ( '' == $author->user_email ) {
```
Comment (mjangda) [inline co-authors-plus.php]: Should we validate using `is_email` as well?

File: co-authors-plus.php [hunk 3]
```diff
@@ -1540,118 +1543,46 @@ public function clear_cache_on_terms_set( $object_id, $terms, $tt_ids, $taxonomy
...
-			}
-			$notify_message .= get_permalink( $comment->comment_post_ID ) . "#comments\r\n\r\n";
-			$notify_message .= sprintf( __( 'Permalink: %s' ), get_permalink( $comment->comment_post_ID ) . '#comment-' . $comment_id ) . "\r\n";
-			if ( EMPTY_TRASH_DAYS ) {
-				$notify_message .= sprintf( __( 'Trash it: %s' ), admin_url( "comment.php?action=trash&c=$comment_id" ) ) . "\r\n";
-			} else {
-				$notify_message .= sprintf( __( 'Delete it: %s' ), admin_url( "comment.php?action=delete&c=$comment_id" ) ) . "\r\n";
-			}
-			$notify_message .= sprintf( __( 'Spam it: %s' ), admin_url( "comment.php?action=spam&c=$comment_id" ) ) . "\r\n";
+if ( ! function_exists ( 'cap_notify_coauthors') ) :
```
Comment (mjangda) [inline co-authors-plus.php]: I think we can move this function inside the class and get rid of the function exists check.

**Comment by rebeccahum** [comment]
Completed the changes as per your request @mjangda.  Let me know if you need anything else.

**Comment by lidocaine** [comment]
I'm writing up some test cases for this PR, but I want to make sure I checked for the right conditions:

1. Default behavior dictates the post author is notified of all comments normally, except for comments they made themselves to their own post.
2. By default, this method should return an array of the post author and all _additional_ coauthors.
3. The returned array should not include coauthors with an invalid email address (malformed, empty, or null).
4. The returned array should not notify the user if they are moderating a comment on their own post (i.e. skip appending their email address to the returned array).

If that is all correct, everything looks good except when considering the condition 1, it seems that the post author is would be notified twice for every comment he did not make or moderate himself. To address this I've added and tested the below check to see if a coauthor's email address is already present in the array passed to the filter, starting on line 1559, right after `$email = is_email( $author->user_email );` in the `foreach`.

```PHP
if ( in_array( $author->user_email, $emails ) ) {
    continue;
}
```

I've also attempted to pull in the the master branch and resolve any conflicts. However, I can't push to this PR directly... @rebeccahum I can either issue a separate PR or provide the contents of my `test-notify-coauthors-filter.php` and altered `co-authors-plus.php`. I've pushed my changes up to my copy of your PR--everything can be seen in [the branch for this PR in my fork of CAP, here](https://github.com/lidocaine/Co-Authors-Plus/tree/pr/421-rebeccahum).

**Comment by GaryJones** [comment]
As this PR contains many irrelevant changes (Travis config, whitespace formatting, version bumps, etc.) and is conflicted and old, I'm going to close it out. Feel free to open a new targeted PR if the change is still desired.

**Pull request closed without merging**